### ISSUE TITLE
The sklearn version of eval.py

### ISSUE THREAD

**Issue by darkstar1227** [WZH0120/SAM2-UNet#45]
As the title, it is the **sklearn** version **eval.py**.
Because the  py_sod_metrics package is to old.
@WZH0120 is that possible I can have a request to refresh your eval.py?

`
import os
import cv2
import argparse
import numpy as np
from sklearn.metrics import f1_score, jaccard_score, mean_absolute_error, precision_score, recall_score, accuracy_score
import concurrent.futures

parser = argparse.ArgumentParser()
parser.add_argument("--dataset_name", type=str, required=True, 
                    help="Dataset name")
parser.add_argument("--pred_path", type=str, required=True, 
                    help="Path to the prediction results")
parser.add_argument("--gt_path", type=str, required=True,
                    help="Path to the ground truth masks")
parser.add_argument("--img_type", type=str, default='jpg',
                    help="Image type of the prediction results")
args = parser.parse_args()

pred_root = args.pred_path
mask_root = args.gt_path
mask_list = sorted(os.listdir(mask_root))
img_type = args.img_type if args.img_type.startswith('.') else '.' + args.img_type

def process_mask(mask_name, idx,img_type='jpg'):
    print(f"[{idx}] Processing {mask_name}...")
    mask_path = os.path.join(mask_root, mask_name)
    pred_path = os.path.join(pred_root, mask_name[:-4] + img_type)
    mask = cv2.imread(mask_path, cv2.IMREAD_GRAYSCALE)
    pred = cv2.imread(pred_path, cv2.IMREAD_GRAYSCALE)

    if mask is None or pred is None:
        print(f"Skipping {mask_name} due to missing file.")
        return None

    # Binarize the image (assuming 127 is the threshold)
    mask_bin = (mask > 127).astype(np.uint8)
    pred_bin = (pred > 127).astype(np.uint8)

    # Convert to 1D array for sklearn calculation
    mask_flat = mask_bin.flatten()
    pred_flat = pred_bin.flatten()

    dice = f1_score(mask_flat, pred_flat, zero_division=0)
    iou = jaccard_score(mask_flat, pred_flat, zero_division=0)
    mae_val = mean_absolute_error(mask_flat, pred_flat)
    precision_val = precision_score(mask_flat, pred_flat, zero_division=0)
    recall_val = recall_score(mask_flat, pred_flat, zero_division=0)
    accuracy_val = accuracy_score(mask_flat, pred_flat)
    
    return dice, iou, mae_val, precision_val, recall_val, accuracy_val

dice_scores = []
iou_scores = []
mae_scores = []
precision_scores = []
recall_scores = []
accuracy_scores = []

with concurrent.futures.ThreadPoolExecutor(max_workers=8) as executor:
    # Building an index with enumerate
    futures = {executor.submit(process_mask, mask_name, i, img_type): mask_name for i, mask_name in enumerate(mask_list)}
    
    for future in concurrent.futures.as_completed(futures):
        result = future.result()
        if result is not None:
            dice, iou, mae_val, precision_val, recall_val, accuracy_val = result
            dice_scores.append(dice)
            iou_scores.append(iou)
            mae_scores.append(mae_val)
            precision_scores.append(precision_val)
            recall_scores.append(recall_val)
            accuracy_scores.append(accuracy_val)

mean_dice = np.mean(dice_scores)
median_dice = np.median(dice_scores)
std_dice = np.std(dice_scores)

mean_iou = np.mean(iou_scores)
median_iou = np.median(iou_scores)
std_iou = np.std(iou_scores)

mean_mae = np.mean(mae_scores)
median_mae = np.median(mae_scores)
std_mae = np.std(mae_scores)

mean_precision = np.mean(precision_scores)
median_precision = np.median(precision_scores)
std_precision = np.std(precision_scores)

mean_recall = np.mean(recall_scores)
median_recall = np.median(recall_scores)
std_recall = np.std(recall_scores)

mean_accuracy = np.mean(accuracy_scores)
median_accuracy = np.median(accuracy_scores)
std_accuracy = np.std(accuracy_scores)

print(args.dataset_name)
print("mDice:      ", format(mean_dice, '.3f'), " Med:", format(median_dice, '.3f'), " Std:", format(std_dice, '.3f'))
print("mIoU:       ", format(mean_iou, '.3f'), " Med:", format(median_iou, '.3f'), " Std:", format(std_iou, '.3f'))
print("MAE:        ", format(mean_mae, '.3f'), " Med:", format(median_mae, '.3f'), " Std:", format(std_mae, '.3f'))
print("Precision:  ", format(mean_precision, '.3f'), " Med:", format(median_precision, '.3f'), " Std:", format(std_precision, '.3f'))
print("Recall:     ", format(mean_recall, '.3f'), " Med:", format(median_recall, '.3f'), " Std:", format(std_recall, '.3f'))
print("Accuracy:   ", format(mean_accuracy, '.3f'), " Med:", format(median_accuracy, '.3f'), " Std:", format(std_accuracy, '.3f'))`

**Comment by congmingyige** [comment]
pred_path = os.path.join(pred_root, mask_name[:-4] + img_type): It is lack of '.'
@darkstar1227

**Comment by congmingyige** [comment]
Why this result is different from the result from "eval.py"?
@darkstar1227
